code/datums/elements/bump_click.dm

```diff
@@ -0,0 +1,75 @@
+/**
+ * Bump click bespoke element
+ *
+ * Simulates a click on the attached atom when it's bumped, only if an item in the bumper's hand slots has a specific tool_behaviour, or
+ * is a specific item or its subtypes or if the active hand slot is empty and empty-handedness is allowed.
+ */
+/datum/element/bump_click
+	element_flags = ELEMENT_BESPOKE
+	id_arg_index = 2
+	///Behaviours to look for in bumper's in-hand objects before attacking the attached atom with one.
+	var/list/tool_behaviours
+	///Tool types to look for in bumper's in-hand objects before attacking the attached atom with one.
+	var/list/tool_items
+	///Do clicks with an empty active hand go through?
+	var/allow_unarmed = FALSE
+	///Do clicks on combat mode go through?
+	var/allow_combat = FALSE
+
+/datum/element/bump_click/Attach(datum/target, list/tool_behaviours, list/tool_items, allow_unarmed = FALSE, allow_combat = FALSE)
+	. = ..()
+
+	if(!isatom(target) || isarea(target))
+		return ELEMENT_INCOMPATIBLE
+	src.tool_behaviours = tool_behaviours
+	src.tool_items = typecacheof(tool_items)
```

Comment:
Typecaches are expensive to generate. s important we only create them once, if we need them, rather then doing it each time the element is added
```suggestion
	if(length(tool_items) && !length(src.tool_items))
		src.tool_items = typecacheof(tool_items)
```